zqd/search/search.go

```diff
@@ -49,27 +45,18 @@ func NewSearch(ctx context.Context, s *space.Space, req api.SearchRequest) (*Sea
 	if err != nil {
 		return nil, err
 	}
-	var f io.ReadCloser
-	f, err = s.OpenFile(space.AllBzngFile)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return nil, err
-		}
-		f = ioutil.NopCloser(strings.NewReader(""))
-	}
-	zngReader, err := detector.LookupReader("bzng", f, resolver.NewContext())
+
+	zngReader, err := s.OpenZng(query.Span)
 	if err != nil {
-		f.Close()
 		return nil, err
 	}
 	zctx := resolver.NewContext()
 	mapper := scanner.NewMapper(zngReader, zctx)
 	mux, err := launch(ctx, query, mapper, zctx)
 	if err != nil {
-		f.Close()
```

Comment:
we still need to close zngReader here

Reply:
whoops, rebase slopiness, thanks for catching that.